pkg/services/ngalert/api/authorization.go

```diff
@@ -97,6 +97,10 @@ func (api *API) authorize(method, path string) web.Handler {
 
 	// Alert Instances and Silences
 
+	// Alert Instance State History. Grafana Path.
+	case http.MethodGet + "/api/v1/ngalert/history/grafana/rules/{UID}":
+		eval = ac.EvalPermission(ac.ActionAlertingRuleRead)
```

Comment:
This authorization is not enough because the permission to access an alert rule is quite complex: 
- action to read alert ruler with folder scope.
- access to all data sources that all rules in the group use.

Usually, we handle the first permission in this function and the second in the controller.